transformations/transformations/single_column_coalesced.py

```diff
@@ -577,9 +632,12 @@ def annotate_driver(cls, directive, driver_loop, kernel_loop, block_dim, column_
 
             # Annotate vector loops with OpenACC pragmas
             if kernel_loop:
-                kernel_loop._update(pragma=ir.Pragma(keyword='acc', content='loop vector'))
+                for loop in kernel_loop:
+                    loop._update(pragma=(ir.Pragma(keyword='acc', content='loop vector'),))
 
-            if driver_loop.pragma is None:
+            if driver_loop.pragma is None or (len(driver_loop.pragma) == 1 and
```

Comment:
Yes, but the additional check is for a different reason. 
The problem is that I've added the driver loop pragma and with the former/original logic this means that the driver loop has a pragma (thus `driver_loop.pragma` is None evaluates to `False`) but this pragma is not a `acc data` statement and consequently the `acc parallel loop gang ...` would not be inserted ... 
I think we should improve the logic here. However, I think for now it's reasonable to do it like this and improve it whenever we add OpenMP pragmas (as alternative to OpenACC pragmas) ?! 

Reply:
Yes, again, good point, and fully agreed. This will need a little logic reshuffling, but it's reasonable to attepmt this later (and possibly once OpenMP-offload is needed for this.

Reply:
This sounds like another use-case where better API to work with pragmas might be useful. We should improve our utilities for that.